tools/clang/lib/SPIRV/DeclResultIdMapper.h

```diff
@@ -437,8 +437,10 @@ class DeclResultIdMapper {
     /// Default constructor to satisfy DenseMap
     DeclSpirvInfo() : instr(nullptr), indexInCTBuffer(-1) {}
 
-    DeclSpirvInfo(SpirvInstruction *instr_, int index = -1)
-        : instr(instr_), indexInCTBuffer(index) {}
+    DeclSpirvInfo(DeclResultIdMapper *declResultIdMapper,
+                  SpirvInstruction *instr_, int index = -1)
+        : instr(declResultIdMapper->getOrCreateCloneVarForFxcCTBuffer(instr_)),
```

Comment:
I'm a bit confused about how you're approaching this. I think it's a bit hacky to call `getOrCreateCloneVarForFxcCTBuffer` for every single Decl. (DeclSpirvInfo is created for every Decl).

Unless I have missed something, I think the following can simplify and clean up the implementation a lot:
* in the SpirvEmitter entry function wrapper, figure out which variables (if any) need to have a clone. Then create a clone for them by calling `declResultIdMapper.createCloneVarForFxcCTBuffer` in SpirvEmitter.
* Don't create varToClone map. You can map from the VarDecl to the clone spir-v instruction using the regular `astDecls` map.
* You shouldn't need most (or all) of the newly added SpirvBuilder functions. You could call the existing SpirvBuilder functions from SpirvEmitter.

Maybe I haven't understood the reasoning for some of the design choices you've made. I'm available to meet if you'd like to go over the details. Thanks!

Reply:
We had an offline discussion. Let me mark it as "resolved".